packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,217 @@
+            // Call the Safe to remove the owner and update the threshold
+            uint256 thresholdAfter = get75PercentThreshold(numOwnersAfter);
+            address prevOwner = _getPrevOwner(owner, owners);
+            _removeOwner({ _prevOwner: prevOwner, _owner: owner, _threshold: thresholdAfter });
+        } else {
+            // The number of owners is dangerously low, so we wish to transfer the ownership of this Safe
+            // to the fallback owner.
+
+            // Remove owners one at a time starting from the last owner.
+            // Since we're removing them in order from last to first, the ordering will remain constant,
+            // and we shouldn't need to query the list of owners again.
+            address prevOwner;
+            for (uint256 i = owners.length - 1; i > 0; i--) {
+                address currentOwner = owners[i];
+                prevOwner = _getPrevOwner(currentOwner, owners);
+
+                // Call the Safe to remove the owner
+                _removeOwner({ _prevOwner: prevOwner, _owner: currentOwner, _threshold: 1 });
+            }
+
+            prevOwner = _getPrevOwner(owners[0], owners);
+            // Add the fallback owner as the sole owner of the Safe
+            _swapToFallbackOwner({ _prevOwner: prevOwner, _oldOwner: owners[0] });
+        }
+
+        _verifyFinalState();
+    }
+
+    /// @notice Sets the fallback owner as the sole owner of the Safe with a threshold of 1
+    /// @param _prevOwner Owner that pointed to the owner to be replaced in the linked list
+    /// @param _oldOwner Owner address to be replaced.
+    function _swapToFallbackOwner(address _prevOwner, address _oldOwner) internal {
+        SAFE.execTransactionFromModule({
+            to: address(SAFE),
+            value: 0,
+            operation: Enum.Operation.Call,
+            data: abi.encodeCall(OwnerManager.swapOwner, (_prevOwner, _oldOwner, FALLBACK_OWNER))
+        });
+    }
+
+    /// @notice Removes the owner `owner` from the Safe and updates the threshold to `_threshold`.
+    /// @param _prevOwner Owner that pointed to the owner to be removed in the linked list
+    /// @param _owner Owner address to be removed.
+    /// @param _threshold New threshold.
+    function _removeOwner(address _prevOwner, address _owner, uint256 _threshold) internal {
+        SAFE.execTransactionFromModule({
+            to: address(SAFE),
+            value: 0,
+            operation: Enum.Operation.Call,
+            data: abi.encodeCall(OwnerManager.removeOwner, (_prevOwner, _owner, _threshold))
+        });
+    }
+
+    /// @notice A FREI-PI invariant check enforcing requirements on number of owners and threshold.
+    function _verifyFinalState() internal view {
+        address[] memory owners = SAFE.getOwners();
+        uint256 numOwners = owners.length;
+        require(
+            _isAboveMinOwners(numOwners) || (numOwners == 1 && owners[0] == FALLBACK_OWNER),
```

Comment:
If this was split into 2 require statements, it would be easier to test both cases.

```solidity
if (numOwners == 1) {
  require(owners[0] == FALLBACK_OWNER , "...");
} else {
  require(_isAboveMinOwners(numOwners), "...");
}
```

Where we can be certain we are hitting both branches in tests by using different require messages

Reply:
There's really only one check happening here though, ie. this rewrites to: 
```
        if (numOwners < MIN_OWNERS) {
            require(
                numOwners == 1 && owners[0] == FALLBACK_OWNER,
                "LivenessModule: Safe must have the minimum number of owners or be owned solely by the fallback owner"
            );
        }
```

I prefer writing it as a single statement indicating the two acceptable states.

Reply:
Good call, I did that, though I don't see a way to test this branch: 

```
if (numOwners == 1) { require(owners[0] == FALLBACK_OWNER , "...");
```

it's difficult to use cheatcodes to intervene, since FALLBACK_OWNER is an immutable.

Reply:
Hmmmm, I seem to have lost that change. Will restore.